scripts/mixins/families/chigoe_pet.lua

```diff
@@ -0,0 +1,96 @@
+-- Chigoe(pet) family mixin
+
+require('scripts/globals/mixins')
+
+g_mixins = g_mixins or {}
+g_mixins.families = g_mixins.families or {}
+
+g_mixins.families.chigoe = function(chigoeMob)
+    -- Chigoes should spawn for the first 5 mobskills
+    chigoeMob:addListener('WEAPONSKILL_USE', 'MOB_SPAWN_CHIGOE', function(mobArg)
+        local chigoeCount = mobArg:getLocalVar('ChigoeCount')
+        local zone        = mobArg:getZone()
+
+        local chigoe = zone:insertDynamicEntity({
+            objtype     = xi.objType.MOB,
+            name        = 'Chigoe',
+            look        = 1747,
+            widescan    = 1,
+            groupId     = 6,
+            groupZoneId = 51,
+
+            onMobSpawn = function(mobArgTwo)
```

Comment:
You can change the naming all around here to be: chigoe -> chigoeDE -> chigoeMob, or something, `mobArgTwo` is pretty unclear, given that this mixin is counting and spawning multiple of this mob as the result of a listener